homeassistant/components/sms/config_flow.py

```diff
@@ -2,17 +2,27 @@
 import logging
 
 import gammu  # pylint: disable=import-error
-import voluptuous as vol
+import voluptuous as vol  # pylint: disable=import-error
 
 from homeassistant import config_entries, core, exceptions
 from homeassistant.const import CONF_DEVICE
+import homeassistant.helpers.config_validation as cv
 
-from .const import DOMAIN
+from .const import (  # pylint: disable=unused-import
+    CONF_BAUD_SPEED,
+    DEFAULT_BAUD_SPEED,
+    DOMAIN,
+)
 from .gateway import create_sms_gateway
 
 _LOGGER = logging.getLogger(__name__)
 
-DATA_SCHEMA = vol.Schema({vol.Required(CONF_DEVICE): str})
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE): str,
+        vol.Optional(CONF_BAUD_SPEED, default=DEFAULT_BAUD_SPEED): cv.positive_int,
```

Comment:
Should this be a list to select from? Feel a bit weird to add a number field for this (e.g., I could enter `38235`) 🤷

Reply:
If we add a list, we could end up in the same situation that we are trying to solve. There are some modems that do not provide automatic speed control and use very specific baud speeds. We should not list them all.
If you enter a value not supported by the modem the configuration will fail and you can retry again with the value that comes from the specific model.

Reply:
Those baud speeds may not be automatic, but still, conform to a standard set?

For example:

- 50
- 75
- 110
- 134
- 150
- 200
- 300
- 600
- 1200
- 1800
- 2400
- 4800
- 9600
- 19200
- 28800
- 38400
- 57600
- 76800
- 115200

I don't think we need to go any higher, but some platforms also support:

- 230400
- 460800
- 576000
- 921600
- etc

Other values may cause issues, and, I didn't see any case that required anything else?

Reply:
I assume this list will work. Can you show me how to specify a list of values in the data schema?

Reply:
import:

`from homeassistant.helpers import selector`

and (unformatted):

```py
        vol.Optional(CONF_BAUD_SPEED, default=DEFAULT_BAUD_SPEED): selector.selector(
            {"select": {"options": [
              {"value": 0, "label": "Auto"},
              {"value": 50, "label": "50"},
              {"value": 75, "label": "75"},
              ...
            ]}}
        ),
```
The whole dict is properly easier to create using comprehension, but this above makes the example/use clear.

Reply:
Awesome, will do

Reply:
I am seeing an error: `expected str for dictionary value @ data['options'][0]['value']`
I think is because is expecting str and I am giving it integers.
How do I specify the type of the selector?

Reply:
> I am seeing an error: `expected str for dictionary value @ data['options'][0]['value']` I think is because is expecting str and I am giving it integers. How do I specify the type of the selector?

I work around the issue by using "string" for the speed value.